core/api/src/graphql/admin/schema.graphql

```diff
@@ -82,8 +82,15 @@ type BTCWallet implements Wallet {
   balance: SignedAmount!
   id: ID!
 
+  """
+  The lightning invoice with the matching paymentHash belonging to this wallet.
+  """
+  invoiceByHash(paymentHash: PaymentHash!): LnInvoice!
+
   """An unconfirmed incoming onchain balance."""
   pendingIncomingBalance: SignedAmount!
+  transactionByHash(hash: PaymentHash!): Transaction!
```

Comment:
could there be multiple transaction by hash?

can hash be only a lightning hash? or could it be something else (like txid?) 

I think some description here would be relevant

there are cases where we can return multiple transaction per hash.

- on lightning, if multiple payment attempt were made (not too sure actually)
- on chain, if address is been reused

Reply:
I've switched this to be `transactionByPaymentHash` so that it only for lightning invoices. This makes it guaranteed to a single transaction. I have also made it nullable.